run-checks

```diff
@@ -334,6 +334,35 @@ if [ "$STATIC" = 1 ]; then
       echo "$got"
       exit 1
     fi
+
+    echo "Checking installation of golangci-lint"
+    GOLANGCI_INSTALL_DIR="/tmp"
+    GOLANGCI_LINT="$GOLANGCI_INSTALL_DIR/golangci-lint"
+    # Ensure GOLANGCI_VERSION matches the golangci-lint version in
+    # .github/workflows/test.yaml (without leading v)
+    GOLANGCI_VERSION="1.51.2"
+    if ! command -v $GOLANGCI_LINT > /dev/null || ! $GOLANGCI_LINT --version | grep -q $GOLANGCI_VERSION ; then
+        if command -v curl > /dev/null ; then
+            curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$GOLANGCI_INSTALL_DIR" "v$GOLANGCI_VERSION"
+        elif command -v wget /dev/null ; then
+            wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$GOLANGCI_INSTALL_DIR" "v$GOLANGCI_VERSION"
+        else
+            echo 'ERROR: could not install golangci-lint. Please do one of the following:'
+            echo '1. install manually from: https://golangci-lint.run/usage/install/#local-installation'
+            echo '2. install curl or wget and re-run this script'
+            exit 1
+        fi
+        if ! command -v $GOLANGCI_LINT > /dev/null || ! $GOLANGCI_LINT --version | grep -q "$GOLANGCI_VERSION" ; then
+            echo "ERROR: installation of golangci-lint proceeded, but executable not found in $GOLANGCI_INSTALL_DIR."
+            echo "Please ensure that golangci-lint v$GOLANGCI_VERSION is installed."
+            echo "Installation instructions are available at:"
+            echo "    https://golangci-lint.run/usage/install/#local-installation"
+            exit 1
+        fi
+    fi
+
+    echo "Running golangci-lint"
+    $GOLANGCI_LINT run || exit 1
```

Comment:
is the exit 1 required?

Reply:
The thought was to exit early on error, rather than continue with other tests, similar to how `go vet` works I believe. Do you think tests should continue on linting failure?